support/cas-server-support-otp-mfa-core/src/main/java/org/apereo/cas/otp/web/flow/OneTimeTokenAccountSaveRegistrationAction.java

```diff
@@ -16,16 +20,39 @@
  * @author Misagh Moayyed
  * @since 5.0.0
  */
+@Slf4j
 @RequiredArgsConstructor
 public class OneTimeTokenAccountSaveRegistrationAction extends AbstractAction {
     private final OneTimeTokenCredentialRepository repository;
+    private final IGoogleAuthenticator googleAuthenticatorInstance;
 
     @Override
     protected Event doExecute(final RequestContext requestContext) {
+        val uid = WebUtils.getAuthentication(requestContext).getPrincipal().getId();
+        val credential = WebUtils.getCredential(requestContext, OneTimeTokenCredential.class);
+
+        if (!StringUtils.isNumeric(credential.getToken())) {
```

Comment:
Not sure on the actual format of the verification code, but `StringUtils.isNumeric` would [return `true` if input contains _Unicode digits_ only](http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNumeric-java.lang.CharSequence-) - including non decimal latin characters (e.g.: `StringUtils.isNumeric("१२३"); // true`).

It also doesn't guarantee that the input is parseable through `Integer.parseInt`.

Reply:
This is another case where the current implementation elsewhere is the same, but I suspect it warrants changing in both locations. I'm working on factoring out the parsing logic to avoid tying gauth-specific code to OTP classes, so can hopefully kill two birds with one stone that way.

Reply:
Updated to just try calling `Integer.parseInt`, passing up an `IllegalArgumentException` on failure, as in `GoogleAuthenticatorAuthenticationHandler`.